src/modules/hash.js

```diff
@@ -0,0 +1,35 @@
+import Crypto from 'crypto-js'
+import { DIMENSION_TYPE_EXPRESSION_DIMENSION_ITEM } from './dataTypes.js'
+import { layoutGetAllItems } from './layout/layoutGetAllItems.js'
+
+export const hash = (value) => Crypto.SHA1(value).toString()
+
+const testValue = (value) => typeof value === 'string' && value.length
```

Comment:
```suggestion
const isValidString = (value) => typeof value === 'string' && value.length
```
I'd probably rename this to something like `isValidString` or sth descriptive, to avoid confusion with "test" (like Cypress or Jest tests) and to indicate that it returns a boolean (which normally starts with "is")

Reply:
Changed to `isValidValue`